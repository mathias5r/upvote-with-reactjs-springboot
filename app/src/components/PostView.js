import React, { Component } from 'react';
import './PostView.css';
import Post from './Post'
import { getPosts } from './API'
import Alert from 'react-s-alert';

class PostView extends Component{

	constructor(props) {
		super(props);
	  	this.state = {
	    	posts: []
	  	};
  	}

  	componentDidMount() {
  		getPosts().then(res =>{
      		const posts = res.data;
     		posts.reverse();
     		this.setState({ posts });
  		}).catch(error =>{
	        Alert.error('Erro ao tentar obter posts!',{
	        	position: 'bottom-right'
	        });
  		})
	}

  	render() {
    	return (
    		<div className="container">
    			<Alert stack={{limit: 2}} />
				{this.state.posts.map(post => (
					<ul className="list-group">
						<div className="jumbotron jumbo">
							<li key={post.id}>
								<Post post={post} />						
							</li>
						</div>
					</ul>
				))}		
			</div>
    	);
  	}
}
export default PostView;

