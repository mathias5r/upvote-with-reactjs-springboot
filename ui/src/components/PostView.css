.container{
	width: 50%;
}

.jumbo {
	margin: 10px;
	padding: 15px;
}


ul {
  list-style-type: none;
}

.text-box {
	background-color: black;
	color: white;
	height: 100%;
	max-height: 200px;
}

.text {
	padding: 10px;
}

.upvotes {
	width: 50px;
	float: right;
	margin: 0 auto;
}

.upvotes-value{
	float: right;
	margin: 0 auto;
	line-height: 50px;
}

.outer{
	width: 100%;
	align-content: right;
	text-align: center;
}

img {
	width: 10%;
}