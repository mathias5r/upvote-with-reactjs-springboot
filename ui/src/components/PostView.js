import React, { Component } from 'react';
import axios from 'axios';
import './PostView.css';
import Post from './Post'

class PostView extends Component{

	constructor(props) {
		super(props);
	  	this.state = {
	    	posts: []
	  	};
  	}

  	componentDidMount() {
	    axios.get('http://localhost:8080/posts').then(res => {
      		const posts = res.data;
      		posts.reverse();
      		this.setState({ posts });
	    });
	}

  	render() {
    	return (
    		<div className="container">
				{this.state.posts.map(post => (
					<ul className="list-group">
						<div className="jumbotron jumbo">
							<li key={post.id}>
								<Post post={post} />						
							</li>
						</div>
					</ul>
				))}		
			</div>
    	);
  	}
}
export default PostView;

