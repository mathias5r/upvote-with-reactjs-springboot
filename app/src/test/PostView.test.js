import React from 'react';
import {shallow} from 'enzyme';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import PostView from '../components/PostView'
import ShallowRenderer from 'react-test-renderer/shallow';;
Enzyme.configure({ adapter: new Adapter() });

describe('Testing PostView.js', () =>{
	it('should render correctly...', () =>{
		const renderer = new ShallowRenderer();
		renderer.render(<PostView />);
	  	const tree = renderer.getRenderOutput();
  		expect(tree).toMatchSnapshot();
	});
});