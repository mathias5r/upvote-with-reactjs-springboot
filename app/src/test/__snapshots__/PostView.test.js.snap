// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing PostView.js should render correctly... 1`] = `
<div
  className="container"
>
  <SAlert
    stack={
      Object {
        "limit": 2,
      }
    }
  />
</div>
`;
